frame/composable-traits/src/staking/mod.rs

```diff
@@ -70,32 +173,27 @@ pub trait Staking {
 	type Balance;
 	type PositionId;
 
-	/// Stake an amount of protocol asset. A new NFT representing the user position will be
-	/// minted.
+	/// Stake an amount of protocol asset.
 	///
 	/// Arguments
 	///
-	/// * `asset` the protocol asset to stake.
-	/// * `from` the account to transfer the stake from.
-	/// * `amount` the amount to stake.
-	/// * `duration` the staking duration (must be one of the predefined presets). Unstaking before
-	///   the end trigger the unstake penalty.
+	/// * `who` the account to transfer the stake from.
+	/// * `amount` the amount to stake. the end trigger the unstake penalty.
 	/// * `keep_alive` whether to keep the `from` account alive or not while transferring the stake.
-	/// * `config_index` - reference to one of  staking configs for `pool_id`
 	fn stake(
 		who: &Self::AccountId,
 		pool_id: &Self::PoolId,
-		config_index: u8,
```

Comment:
how would user choose possible lock config our of several possible then when he stakes?

Reply:
Makes sense, but the name wasn't clear to me. Let me make a change. Thanks